src/libraries/System.Private.CoreLib/src/System/Int16.cs

```diff
@@ -972,6 +978,13 @@ private static bool TryConvertFromSaturating<TOther>(TOther value, out short res
                          (actualValue <= BitConverter.UInt16BitsToHalf(0xF800)) ? MinValue : (short)actualValue;
                 return true;
             }
+            else if (typeof(TOther) == typeof(BFloat16))
+            {
+                BFloat16 actualValue = (BFloat16)(object)value;
+                result = (actualValue >= new BFloat16(0x4700)) ? MaxValue :
+                         (actualValue <= new BFloat16(0xB700)) ? MinValue : (short)actualValue;
```

Comment:
We should have and use an internal `UInt16BitsToBFloat16` for parity with the other floating-point support.